pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -3624,6 +3624,55 @@ public PinotResourceManagerResponse dropInstance(String instanceName) {
     return PinotResourceManagerResponse.success("Instance " + instanceName + " dropped");
   }
 
+  /**
+   * Drains a minion instance by preventing new task assignments while allowing existing tasks to complete.
+   * This is achieved by changing the instance tag from minion_untagged to minion_drained, which prevents
+   * Helix from assigning new tasks to this instance.
+   *
+   * @param instanceName Name of the minion instance to drain
+   * @return Response indicating success or failure
+   */
+  public synchronized PinotResourceManagerResponse drainMinionInstance(String instanceName) {
+    InstanceConfig instanceConfig = getHelixInstanceConfig(instanceName);
+    if (instanceConfig == null) {
+      return PinotResourceManagerResponse.failure("Instance " + instanceName + " not found");
+    }
+
+    // Replace minion_untagged with minion_drained tag
+    List<String> updatedTags = replaceMinionTag(instanceConfig.getTags());
```

Comment:
we should check, if there is any tag other than `minion-untagged` or default tag, then we should just return unsupport exception